src/Stats.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+namespace reactivesocket {
+// forward decl for NoopStats
+class NoopStats;
+
+// Virtual interface for a stats receiver
+class Stats {
+ public:
+  static NoopStats noop;
```

Comment:
Yep, static in a function should be fine. Although I find it hard to imagine a usage of this static that'd lead to SOIF...